airflow/jobs/local_task_job.py

```diff
@@ -264,3 +265,15 @@ def _run_mini_scheduler_on_child_tasks(self, session=None) -> None:
                 exc_info=True,
             )
             session.rollback()
+
+    @provide_session
+    def _update_dagrun_state_for_paused_dag(self, session=None):
+        """
+        Checks for paused dags with DagRuns in the running state and
+        update the DagRun state if possible
+        """
+        dag_run = self.task_instance.get_dagrun(session=session)
+        if dag_run:
+            dag = dag_run.dag = self.task_instance.task.dag
+            if dag.get_is_paused(session=session):
+                dag_run.update_state(session=session, execute_callbacks=True)
```

Comment:
We can delay getting the dag_run until after we've checked if the dag is paused (one less query) by using `self.task_instances.dag_model.is_paused`

Reply:
Done. The self.task_instance.dag_model was returning None for dag_model.  I then changed to self.task_instance.task.dag.